Orange/widgets/visualize/owmosaic.py

```diff
@@ -462,8 +462,14 @@ def handleNewSignals(self):
             else:
                 indices = {e.id for e in transformed}
                 self.subset_indices = [ex.id in indices for ex in self.data]
+        if self.data is not None and self.__pending_selection is not None:
+            self.selection = self.__pending_selection
+            self.__pending_selection = None
+        else:
+            self.selection = set()
         self.set_color_data()
-        self.reset_graph()
+        self.update_graph()
+        self.send_selection()
```

Comment:
@VesnaT, copying from projection widget indeed helped (and you were right about not using `pack_settings`, I should have listened more carefully and be less self confident :)).

However, every widget has also its own tricks, so you can't *just copy*. In this case, it was important to call `send_selection` after `update_graph`.